src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.constructor.alive.push(this);
+  }
+
+  static removeAnimal() {
+    this.alive = this.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal instanceof Carnivore) && !animal.hidden) {
```

Comment:
I'd suggest to check if animal is a herbivore - instead of checking if it's not a carnivore. According to task conditions, this is the feature of herbivores - not everyone except carnivores.
This is the same when you only have two classes, but the difference appears when potentially third class is added.